### PR TITLE
fix: only keep latest failed data directory

### LINKED ISSUE #1421: Standby failed bootstrap fills up disk with timestamped directories
Hi,

We've got a standby cluster where we're clearly doing something wrong in our bootstrap script (a wal-g hack of your wal-e one), but on failure the postgres/current directory is renamed with a timestamp, and the process starts again.  This soon fills up a disk.

Is there any reason not to just rename once to `current.failed` as if this is a repeated problem then only the last failure is really relevant?

Thanks.

### PR DESCRIPTION
**Pull request by walbertus**
Use constant postfix when moving data directory due to failure so it only keeps data from the latest failure.
I need some input, especially on the naming convention.

Close #1421

### CODE REVIEW

File: patroni/postgresql/__init__.py [hunk 1]
```diff
@@ -1023,27 +1023,33 @@ def pg_tblspc_realpaths(self):
...
+                postfix = 'failed'
 
                 # let's see if the wal directory is a symlink, in this case we
                 # should move the target
                 for (source, pg_wal_realpath) in self.pg_wal_realpath().items():
                     logger.info('renaming WAL directory and updating symlink: %s', pg_wal_realpath)
-                    new_name = '{0}_{1}'.format(pg_wal_realpath, postfix)
+                    new_name = '{0}.{1}'.format(pg_wal_realpath, postfix)
+                    if os.path.exists(new_name):
+                        os.remove(new_name)
```
Comment (CyberDem0n) [inline patroni/postgresql/__init__.py:1034]: We better use `shutil.rmtree()` instead of `os.remove()`.

Comment (walbertus) [reply patroni/postgresql/__init__.py]: updated using `shutil.rmtree`

**Comment by walbertus** [comment]
sorry for the late update, added the test to cover the new lines

**Comment by walbertus** [comment]
updated the branch to include codacy fix commit

**Comment by hughcapet** [comment]
👍

**Comment by CyberDem0n** [comment]
👍

**Comment by hughcapet** [comment]
Merged, thank you, @walbertus!

**Pull request merged**